pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/SingularAOMAddParticipantUpdates.java

```
@@ -0,0 +1,509 @@
+            throw new DDPException(message);
+        } else {
+            return true;
+        }
+    }
+
+    private void processStudyQuestionBlockPatch(@NonNull Handle handle,
+                                                @NonNull String studyGuid,
+                                                @NonNull String activityCode,
+                                                @NonNull String versionTag,
+                                                @NonNull String insertAfterStableId,
+                                                @NonNull List<? extends Config> formBlockDefs) {
+        assert StringUtils.isNotBlank(studyGuid);
+        assert formBlockDefs.isEmpty() == false;
+
+        final var daoStudy = handle.attach(JdbiUmbrellaStudy.class);
+        final var daoActivity = handle.attach(JdbiActivity.class);
+        final var daoActivityVersion = handle.attach(JdbiActivityVersion.class);
+        final var daoForm = handle.attach(FormActivityDao.class);
+        final var daoSectionBlock = handle.attach(SectionBlockDao.class);
+        final var sqlHelper = handle.attach(SqlHelper.class);
+
+        final var studyDto = Optional.ofNullable(daoStudy.findByStudyGuid(studyGuid))
+                .orElseThrow(() -> studyNotFoundError(studyGuid));
+
+        final var activity = daoActivity.findActivityByStudyGuidAndCode(studyGuid, activityCode)
+                .orElseThrow(() -> activityNotFoundError(activityCode, studyGuid));
+
+        final var activityVersion = daoActivityVersion.findByActivityCodeAndVersionTag(studyDto.getId(), activityCode, versionTag)
+                .orElseThrow(() -> activityVersionNotFoundError(versionTag, activityCode, studyGuid));
+
+        final var currentActivityDef = daoForm.findDefByDtoAndVersion(activity, activityVersion);
+
+        final var indexBlock = findBlocksForQuestions(currentActivityDef, List.of(insertAfterStableId)).stream()
+                .findFirst()
```

I am not sure about the structure of Questions in DSS, is it a safe assumption that `findFirst()` is enough here, rather than finding the index of all possible blocks?

---

The use of `findFirst()` is intentional here- the code assumes the constraints:
* stable ids are guaranteed to be unique within an activity
* a question has exactly one parent block

I'm not away of any studies which violate these two constraints and, as far as I know, they are enforced by the database schema. If those two constraints don't hold, I want this to fail loudly.